### PR TITLE
Remove support for ROCm < 5.6.0

### PR DESCRIPTION
**Pull request by atamazov**
After recent changes to MIOpen, it cannot be built with ROCm < 5.7.0. Let's remove the useless code.

----
[Attribution] @junliume @JehandadKhan
- https://github.com/ROCm/MIOpen/labels/urgency_normal
- Proposed reviewers: @junliume @JehandadKhan

### CODE REVIEW

**Review by JehandadKhan — APPROVED** [review]

File: src/hip/handlehip.cpp [hunk 1]
```diff
@@ -57,24 +56,17 @@
 #include <mutex>
 #include <shared_mutex>
 
-#define MIOPEN_WORKAROUND_ROCM_COMPILER_SUPPORT_ISSUE_30 \
-    (MIOPEN_USE_COMGR && BUILD_SHARED_LIBS && (HIP_PACKAGE_VERSION_FLAT < 4003000000ULL))
-
+/// GFX906 and GFX103X are deprecated since 5.7 RC.
```
Comment (V6ser) [inline src/hip/handlehip.cpp]: Why exactly was GFX103X deprecated?

Comment (atamazov) [reply src/hip/handlehip.cpp]: @V6ser Not in MIOpen, but in HIP runtime. I have no idea why. It still works, but I see some quirks like this one.

Comment (junliume) [reply src/hip/handlehip.cpp]: @atamazov I would recommend remove or revise this line. It's a bug and issue on hip runtime nevertheless.

Comment (atamazov) [reply src/hip/handlehip.cpp]: @junliume I agree. Please expect update soon.

**Comment by junliume** [comment]
@atamazov it seems that FIN updates are needed for this PR too:
```
[2024-01-13T07:51:06.284Z] /var/jenkins/workspace/MLLIBS_MIOpen_PR-2665/src/ocl/gcn_asm_utils.cpp:181:5: error: 'ROCM_FEATURE_ASM_REQUIRES_NO_XNACK_OPTION' is not defined, evaluates to 0 [-Werror,-Wundef]

[2024-01-13T07:51:06.284Z] #if ROCM_FEATURE_ASM_REQUIRES_NO_XNACK_OPTION

[2024-01-13T07:51:06.284Z]     ^
```

**Comment by atamazov** [comment]
@junliume @cderb 
> @atamazov it seems that FIN updates are needed for this PR too:

sudden fin
image

**Comment by atamazov** [comment]
@junliume BTW this issue was not related to fin. Fixed! Ready for CI testing and review.

**Review by junliume — APPROVED** [review]

**Review by CAHEK7 — APPROVED** [review]

**Comment by junliume** [comment]
@atamazov could you resolve the above conflicts? Thanks!

**Comment by atamazov** [comment]
@junliume Done! Ready for CI testing and review.

**Pull request merged**